### PR TITLE
Add GitHub actions

### LINKED ISSUE #35: Explore using github actions
Maintaining appveyor + travis is quite a bit painful. Github Actions can be a solution. If it is feasible let's move to that.

### PR DESCRIPTION
**Pull request by vertexclique**
Fixes #35

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --all --benches --bins --examples --tests
+
+      - name: tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all --doc
```
Comment (JohnTitor) [inline .github/workflows/ci.yml]: Run doc-tests only?

Comment (vertexclique) [reply .github/workflows/ci.yml]: But tests under the `tests` dir are very important. Checks for some api like shutdown.

Comment (JohnTitor) [reply .github/workflows/ci.yml]: I think we can just use `--all`

Comment (vertexclique) [reply .github/workflows/ci.yml]: ah yes better.

**Pull request merged**